packages/endo/src/cli.js

```diff
@@ -0,0 +1,130 @@
+/* global harden */
+
+import "./lockdown.js";
+import { loadLocation, writeArchive, loadArchive } from "./main.js";
+import { search } from "./search.js";
+import { compartmentMapForNodeModules } from "./compartmap.js";
+
+const iterate = sequence => sequence[Symbol.iterator]();
+
+function usage(message) {
+  console.error(message);
+  return 1;
+}
+
+async function execute(application) {
+  const endowments = harden({
+    console: {
+      log(...args) {
+        console.log(...args);
+      }
+    }
+  });
+
+  await application.execute(endowments);
+  return 0;
+}
+
+async function executeLocation(applicationPath, { cwd, read }) {
+  const currentLocation = new URL(`${cwd()}/`, "file:///");
+  const applicationLocation = new URL(applicationPath, currentLocation);
+  const application = await loadLocation(read, applicationLocation);
+  return execute(application);
+}
+
+async function archive(archivePath, applicationPath, { cwd, read, write }) {
+  const currentLocation = new URL(`${cwd()}/`, "file:///");
+  const archiveLocation = new URL(archivePath, currentLocation);
+  const applicationLocation = new URL(applicationPath, currentLocation);
+  await writeArchive(write, read, archiveLocation, applicationLocation);
+  return 0;
+}
+
+async function executeArchive(archivePath, { read, cwd }) {
+  const currentLocation = new URL(`${cwd()}/`, "file:///");
+  const archiveLocation = new URL(archivePath, currentLocation);
+  const application = await loadArchive(read, archiveLocation);
+  return execute(application);
+}
+
+async function compartmap(applicationPath, { read, cwd }) {
+  const currentLocation = new URL(`${cwd()}/`, "file:///");
+  const applicationLocation = new URL(applicationPath, currentLocation);
+  const { packageLocation } = await search(read, applicationLocation);
+  const compartmentMap = await compartmentMapForNodeModules(
+    read,
+    packageLocation
+  );
+  console.log(JSON.stringify(compartmentMap, null, 2));
```

Comment:
I suggest passing `process.stdout` or `process.stdout.write` to `compartmap()` and using that here instead of `console.log`.

We make an ocap discipline exception and allow ambient access to `console`, but I don't think code should rely on it to do anything; that is: if writing to `stdout` is a feature and not just a diagnostic aid, use `process.stdout` as an explicit capability.